### PR TITLE
initial code drop

### PR DESCRIPTION
**Pull request by InformatiQ**
cookbook to  install bamboo form source and configure it

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+source 'https://rubygems.org'
+
+gem 'test-kitchen'
+gem 'berkshelf'
+gem 'kitchen-vagrant'
+gem 'kitchen-vagrant_winrm'
```
Comment (Annih) [inline Gemfile]: You don't really need kitchen-vagrant_winrm if you test only on centos

File: attributes/default.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+default['qubit_bamboo']['config']['marathon_port'] = 8080
+default['qubit_bamboo']['config']['marathon_hosts'] = ["http://localhost:#{node['qubit_bamboo']['config']['marathon_port']}"]
+
+default['qubit_bamboo']['config']['port'] = 8000
+
+default['qubit_bamboo']['config']['zk']['hosts'] = ['localhost']
+default['qubit_bamboo']['config']['zk']['path'] = ''
+default['qubit_bamboo']['config']['zk']['reporting_delay'] = 5
+
+default['qubit_bamboo']['config']['haptoxy']['template_path'] = "#{node['qubit_bamboo']['home']}/haproxy_template.cfg"
```
Comment (rveznaver) [inline attributes/default.rb]: typo: 'haptoxy' -> 'haproxy'

File: recipes/install_from_source.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#
+# Cookbook Name:: qubit-bamboo
+# Recipe:: default
+#
+# Copyright (c) 2015 The Authors, All Rights Reserved.
+#
+yum_repository 'epel' do
```
Comment (rveznaver) [inline recipes/install_from_source.rb]: Why not just include_recipe 'yum-epel' ?

File: attributes/default.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+default['qubit_bamboo']['config']['haproxy']['output_path'] = '/etc/haproxy/haproxy.cfg'
+default['qubit_bamboo']['config']['haproxy']['ReloadCommand'] = 'haproxy -f /etc/haproxy/haproxy.cfg -p /var/run/haproxy.pid -D -sf $(cat /var/run/haproxy.pid)'
+
+default['qubit_bamboo']['config']['statsd']['enabled'] = false
+default['qubit_bamboo']['config']['statsd']['host'] = 'localhost'
+default['qubit_bamboo']['config']['statsd']['port'] = 8125
+default['qubit_bamboo']['config']['statsd']['prefix'] = 'bamboo'
+
+default['qubit_bamboo']['config'] = {
```
Comment (rveznaver) [inline attributes/default.rb:26]: OPT: IMO, all options set under this hash do not need to be set somewhere else before, as it is quite easy to overwrite attributes as needed.

File: recipes/default.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#
+# Cookbook Name:: qubit-bamboo
+# Recipe:: default
+#
+# Copyright (c) 2015 The Authors, All Rights Reserved.
+#
+
+include_recipe 'qubit-bamboo::install_from_source'
+
+include_recipe 'qubit-bamboo::config'
```
Comment (rveznaver) [inline recipes/default.rb:22]: Missing recipe to define and start up the upstart service.

Comment (InformatiQ) [inline recipes/default.rb:22]: included in the config recipe

File: recipes/install_from_source.rb [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  recursive true
+end
+
+bash 'build_bamboo' do
+  cwd "#{Chef::Config[:file_cache_path]}"
+  code <<-EOF
+  [ -l #{go_link}/bamboo ] || ln -s $(pwd)/bamboo-#{node['qubit_bamboo']['version']}  #{go_link}/bamboo
+  export GOPATH=#{gopath}
+  cd bamboo-#{node['qubit_bamboo']['version']}
+  go build
```
Comment (rveznaver) [inline recipes/install_from_source.rb]: How about `GOPATH=#{gopath} go build` so we don't litter the environment?

File: README.md [hunk 7]
```diff
@@ -2,4 +2,3 @@
 
 TODO: Enter the cookbook description here.
```
Comment (rveznaver) [inline README.md]: Please put in at least one line of description.

File: metadata.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+name             'qubit-bamboo'
+maintainer       'Ramez Hanna'
+maintainer_email 'r.hanna@criteo.com'
+license          'all_rights'
+description      'Installs/Configures qubit-bamboo'
+long_description 'Installs/Configures qubit-bamboo'
+version          '0.1.0'
+
```
Comment (rveznaver) [inline metadata.rb]: Please add `supports` field for centos.

File: recipes/install_from_source.rb [hunk 9]
```diff
@@ -0,0 +1,65 @@
+#
+# Cookbook Name:: qubit-bamboo
+# Recipe:: default
+#
+# Copyright (c) 2015 The Authors, All Rights Reserved.
```
Comment (rveznaver) [inline recipes/install_from_source.rb]: Please include a LICENSE file and set the copyright to Criteo.

File: recipes/install_from_source.rb [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+end
+
+gopath = ::File.join(Chef::Config[:file_cache_path], 'go')
+go_link = ::File.join(gopath, 'src', 'github.com', 'QubitProducts')
+directory go_link do
+  recursive true
+end
+
+bash 'build_bamboo' do
+  cwd "#{Chef::Config[:file_cache_path]}"
```
Comment (rveznaver) [inline recipes/install_from_source.rb]: You can get rid of the string interpolation: `cwd Chef::Config[:file_cache_path]`

File: recipes/install_from_source.rb [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  recursive true
+end
+
+gopath = ::File.join(Chef::Config[:file_cache_path], 'go')
+go_link = ::File.join(gopath, 'src', 'github.com', 'QubitProducts')
+directory go_link do
+  recursive true
+end
+
+bash 'build_bamboo' do
```
Comment (rveznaver) [inline recipes/install_from_source.rb]: This recipe will build from source and copy over the new binary every single time. We should only build once.

File: templates/default/bamboo-server.conf.erb [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+respawn
+respawn limit 2 5
+
+
+umask 007
+
+kill timeout 300
+
+script
+    exec <%= node['qubit_bamboo']['home'] -%>/bamboo -config <%= node['qubit_bamboo']['config_path']-%> -log <%= node['qubit_bamboo']['log_path'] %>
```
Comment (rveznaver) [inline templates/default/bamboo-server.conf.erb]: Does bamboo need to run as root?

File: templates/default/bamboo-server.conf.erb [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+respawn
+respawn limit 2 5
+
+
+umask 007
+
+kill timeout 300
+
+script
```
Comment (rveznaver) [inline templates/default/bamboo-server.conf.erb]: script is not needed if only one exec inside

**Comment by rveznaver** [comment]
+1

**Pull request merged**